pkgs/cronet_http/lib/cronet_client.dart

```diff
@@ -138,11 +177,22 @@ class CronetClient extends BaseClient {
 
   @override
   Future<StreamedResponse> send(BaseRequest request) async {
-    try {
-      _engine ??= await CronetEngine.build();
-    } catch (e) {
-      throw ClientException(e.toString(), request.url);
+    if (_engine == null) {
+      try {
+        if (_engineFuture == null) {
```

Comment:
It's possible to arrive here twice, and construct two different engines (one gets lost), if there are two non-serial calls to `send`.

Consider using `AsyncMemoizer` https://pub.dev/documentation/async/latest/async/AsyncMemoizer-class.html

```dart
  late final _ownEngineMemo = AsyncMemoizer<CronetEngine>();

          _engine = await _ownEngineMemo.runOnce(CronetEngine.build);
```